runtime/drive_handler.go

```diff
@@ -124,6 +124,39 @@ func (h *StubDriveHandler) Reserve(
 	), nil
 }
 
+// Release unmounts stub drive of just deleted container
+// and pushes just released drive to freeDrives
+func (h *StubDriveHandler) Release(
+	requestCtx context.Context,
+	taskID string,
+	driveMounter drivemount.DriveMounterService,
+	machine firecracker.MachineIface,
+) error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	drive, ok := h.usedDrives[taskID]
+	if !ok {
+		return errors.Errorf("container %s drive wasn't found", taskID)
+	}
+
+	_, err := driveMounter.UnmountDrive(requestCtx, &drivemount.UnmountDriveRequest{
```

Comment:
So far, the handler was only doing bookkeeping but now it actually modify a microVM. I'm unsure this is the direction we want to go. Want to know @sipsma and others' views.

Reply:
In practice, what happens now is [you `Reserve` the drive and then immediately call `PatchAndMount`, which takes the same `driveMounter` and `machine` interfaces](https://github.com/firecracker-microvm/firecracker-containerd/blob/802c35dac582ce5543ba6df7e15c215a830c5984/runtime/service.go#L813-L826). So whether you provide `driveMounter` and `machine` to the stub drive handler or a separate method is kind of arbitrary to me.

However, I do think it may be a good idea to leave `Release` as is and update `Reserve` to also take in `driveMounter` and `machine` and directly make the `PatchAndMount` call within its implementation. That not only makes the API a bit more "symmetrical", it also means that we can put the drive in `usedDrives` only when `PatchAndMount` actually succeeds, which reduces the cases where we can enter a state where a stub drive can't be released.